module/bodh/helper.js

```diff
@@ -908,4 +910,175 @@ module.exports = class BodhHelper {
         })
     }
 
+    /**
+      * Create content for platform
+      * @method
+      * @name createContent
+      * @param {String} requestedData
+      * @param {String} token 
+      * @returns {Object} Return content id. 
+     */
+
+    static createContent( requestedData,token ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let contentData = {
+                    "request" : {
+                        "content" : requestedData
+                    }
+                };
+
+                let createdContentData = 
+                await sunbirdService.createContent(
+                    contentData,
+                    token
+                );
+
+                if( createdContentData.responseCode !== constants.common.OK ) {
+                    
+                    throw {
+                        status : httpStatusCode.bad_request.status,
+                        message : createdContentData.result.messages[0]
+                    }
+                }
+
+                return resolve({
+                    message :  constants.apiResponses.CREATED_BODH_CONTENT,
+                    result : {
+                        contentId : createdContentData.result.content_id
+                    }
+                });
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+     /**
+      * Upload scrom content for platform
+      * @method
+      * @name uploadScromContent
+      * @param {String} file - required file
+      * @param {String} contentId - content id
+      * @param {String} token 
+      * @returns {Object} - Return content url.  
+     */
+
+    static uploadScromContent( file,name,token,userId ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let contentCreateData = {
+                    code : gen.utils.generateUniqueId(),
+                    contentType : constants.common.BODH_CONTENT_TYPE,
+                    name : name,
+                    mimeType : constants.common.BODH_MIME_TYPE,
+                    createdBy : userId
+                };
+
+                let contentCreationData = await this.createContent(
+                    contentCreateData,
+                    token
+                );
+
+                let contentId = contentCreationData.result.contentId;
+
+                let saveRequestedZipFile = await filesHelper.saveZipFile(
```

Comment:
@aman-tunerlabs - We should perform all the file check first e.g. extraction of the zip, rename and zip back, etc first and then go for content creation. So that any errors if any are caught in first before reaching out to Sunbird.